plugins/inputs/redis_sentinel/redis_sentinel.go

```diff
@@ -0,0 +1,503 @@
+
+		sentinelMastersTags, sentinelMastersFields, err := convertSentinelMastersOutput(client.baseTags(), m, quorumErr)
+		if err == nil {
+			acc.AddFields(measurementMasters, sentinelMastersFields, sentinelMastersTags)
+		} else {
+			acc.AddError(err)
+		}
+
+		gatherReplicaStats(acc, client, masterName)
+		gatherSentinelStats(acc, client, masterName)
+	}
+}
+
+func gatherReplicaStats(
+	acc telegraf.Accumulator,
+	client *RedisSentinelClient,
+	masterName string,
+) {
+	replicasCmd := redis.NewSliceCmd("sentinel", "replicas", masterName)
+	// We check the command result below
+	//nolint:errcheck
+	client.sentinel.Process(replicasCmd)
+
+	replicas, replicasErr := replicasCmd.Result()
+	if replicasErr != nil {
+		acc.AddError(replicasErr)
+		return
+	}
+
+	for _, replica := range replicas {
+		if replica, replicaOk := replica.([]interface{}); replicaOk {
+			rm := toMap(replica)
+
+			replicaTags, replicaFields, err := convertSentinelReplicaOutput(client.baseTags(), masterName, rm)
+			if err == nil {
+				acc.AddFields(measurementReplicas, replicaFields, replicaTags)
+			} else {
+				acc.AddError(err)
+			}
+		}
+	}
+}
+
+func gatherSentinelStats(
+	acc telegraf.Accumulator,
+	client *RedisSentinelClient,
+	masterName string,
+) {
+	sentinelsCmd := redis.NewSliceCmd("sentinel", "sentinels", masterName)
+	// We check the command result below
+	//nolint:errcheck
+	client.sentinel.Process(sentinelsCmd)
+
+	sentinels, sentinelsErr := sentinelsCmd.Result()
+	if sentinelsErr != nil {
+		acc.AddError(sentinelsErr)
+		return
+	}
+
+	for _, sentinel := range sentinels {
+		if sentinel, sentinelOk := sentinel.([]interface{}); sentinelOk {
+			sm := toMap(sentinel)
+
+			sentinelTags, sentinelFields, err := convertSentinelSentinelsOutput(client.baseTags(), masterName, sm)
+			if err == nil {
+				acc.AddFields(measurementSentinels, sentinelFields, sentinelTags)
+			} else {
+				acc.AddError(err)
+			}
+		}
+	}
+}
+
+// converts `sentinel masters <name>` output to tags and fields
+func convertSentinelMastersOutput(
+	globalTags map[string]string,
+	master map[string]string,
+	quorumErr error,
+) (map[string]string, map[string]interface{}, error) {
+	tags := make(map[string]string)
+	for k, v := range globalTags {
+		tags[k] = v
+	}
+
+	tags["master"] = master["name"]
+
+	fields := make(map[string]interface{})
+
+	fields["has_quorum"] = false
+	if quorumErr == nil {
+		fields["has_quorum"] = true
+	}
+
+	for key, val := range master {
+		key = strings.Replace(key, "-", "_", -1)
+
+		switch valType := measurementMastersFields[key]; valType {
+		case "float":
+			val, err := strconv.ParseFloat(val, 64)
+			if err != nil {
+				return nil, nil, fmt.Errorf("failed parsing field %v: %v", key, err)
+			}
+			fields[key] = val
+		case "integer":
+			val, err := strconv.ParseInt(val, 10, 64)
+			if err != nil {
+				return nil, nil, fmt.Errorf("failed parsing field %v: %v", key, err)
+			}
+			fields[key] = val
+		case "string":
+			fields[key] = val
+		}
+	}
```

Comment:
I think this reoccurs in the other convert function, so you maybe should factor that out to another function....

Reply:
@srebhan Yeah. There was a lot of duplicated code surrounding that logic, so I did some refactoring work in https://github.com/influxdata/telegraf/pull/10042/commits/e6e8cb688147ec560dd7c45fa0cf063c6ab3fe4a.